pyiron_atomistics/atomistics/structure/atoms.py

```diff
@@ -2561,22 +2561,71 @@ def get_initial_magnetic_moments(self):
             else:
                 return np.zeros(len(self))
 
-    def set_initial_magnetic_moments(self, magmoms=None):
+    def set_initial_magnetic_moments(self, magmoms):
         """
         Set array of initial magnetic moments.
 
         Args:
-            magmoms (numpy.ndarray/list): List of magneric moments
+            magmoms (numpy.ndarray/list/dict/float): List, dict or single value
+                of magneric moments
+
+
+        Example I:
+
+        >>> from pyiron_atomistics import Project
+        >>> structure = Project('.').create.structure.bulk('Ni', cubic=True)
+        >>> structure[-1] = 'Fe'
+        >>> v_dict = {'Fe': 1, 'Ni': 2}
+        >>> structure.set_initial_magnetic_moments(v_dict)
+        >>> print(structure.get_initial_magnetic_moments())
+
+        Output:
+
+        >>> array([2, 2, 2, 1])
```

Comment:
```suggestion
        >>> structure.get_initial_magnetic_moments()
        array([2, 2, 2, 1])
```

then it should work directly with the doctest.